src/python/nimbusml/pipeline.py

```diff
@@ -1693,6 +1695,122 @@ def getn(n):
                     "only fit(X) is allowed or the training becomes "
                     "ambiguous.")
 
+    @trace
+    def calculate_feature_contributions(self, X, y=None,
+                 evaltype='auto', group_id=None,
+                 weight=None,
+                 verbose=0,
+                 top=10,
+                 bottom=10,
+                 normalize=True,
+                 as_binary_data_stream=False, **params):
+        """
+        Apply transforms and test with the final estimator, return metrics
+        """
+        # start the clock!
+        start_time = time.time()
+        self.verbose = verbose
+
+        if not self._is_fitted:
+            raise ValueError(
+                "Model is not fitted. Train or load a model before test("
+                ").")
+
+        if y is not None:
+            if len(self.steps) > 0:
+                last_node = self.last_node
+                if last_node.type == 'transform':
+                    raise ValueError(
+                        "Pipeline needs a trainer as last step for test()")
```

Comment:
remove ?

Reply:
This is needed because the last node needs to be a model in order to get feature contributions

---
In reply to: [303648847](https://github.com/microsoft/NimbusML/pull/186#discussion_r303648847) [](ancestors = 303648847)

Reply:
what happens if you load model from file? 

---
In reply to: [303672160](https://github.com/microsoft/NimbusML/pull/186#discussion_r303672160) [](ancestors = 303672160,303648847)

Reply:
Same as with self._is_fitted. Works fine with pickle, but when loading from file self.last_node not set because "TypeError: No steps given."

---
In reply to: [303674473](https://github.com/microsoft/NimbusML/pull/186#discussion_r303674473) [](ancestors = 303674473,303672160,303648847)

Reply:
Works with model loaded from zip file too.

---
In reply to: [303677982](https://github.com/microsoft/NimbusML/pull/186#discussion_r303677982) [](ancestors = 303677982,303674473,303672160,303648847)